testdata/forget_leader.txt

```diff
@@ -0,0 +1,423 @@
+  3->4 MsgApp Term:2 Log:1/11 Commit:11 Entries:[2/12 EntryNormal ""]
+> 1 receiving messages
+  3->1 MsgApp Term:2 Log:1/11 Commit:11 Entries:[2/12 EntryNormal ""]
+> 2 receiving messages
+  3->2 MsgApp Term:2 Log:1/11 Commit:11 Entries:[2/12 EntryNormal ""]
+> 4 receiving messages
+  3->4 MsgApp Term:2 Log:1/11 Commit:11 Entries:[2/12 EntryNormal ""]
+  INFO 4 [term: 1] received a MsgApp message with higher term from 3 [term: 2]
+  INFO 4 became follower at term 2
+> 1 handling Ready
+  Ready MustSync=true:
+  Lead:3 State:StateFollower
+  Entries:
+  2/12 EntryNormal ""
+  Messages:
+  1->3 MsgAppResp Term:2 Log:0/12
+> 2 handling Ready
+  Ready MustSync=true:
+  Lead:3 State:StateFollower
+  Entries:
+  2/12 EntryNormal ""
+  Messages:
+  2->3 MsgAppResp Term:2 Log:0/12
+> 4 handling Ready
+  Ready MustSync=true:
+  Lead:3 State:StateFollower
+  HardState Term:2 Commit:11
+  Entries:
+  2/12 EntryNormal ""
+  Messages:
+  4->3 MsgAppResp Term:2 Log:0/12
+> 3 receiving messages
+  1->3 MsgAppResp Term:2 Log:0/12
+  2->3 MsgAppResp Term:2 Log:0/12
+  4->3 MsgAppResp Term:2 Log:0/12
+> 3 handling Ready
+  Ready MustSync=false:
+  HardState Term:2 Vote:3 Commit:12
+  CommittedEntries:
+  2/12 EntryNormal ""
+  Messages:
+  3->1 MsgApp Term:2 Log:2/12 Commit:12
+  3->2 MsgApp Term:2 Log:2/12 Commit:12
+  3->4 MsgApp Term:2 Log:2/12 Commit:12
+> 1 receiving messages
+  3->1 MsgApp Term:2 Log:2/12 Commit:12
+> 2 receiving messages
+  3->2 MsgApp Term:2 Log:2/12 Commit:12
+> 4 receiving messages
+  3->4 MsgApp Term:2 Log:2/12 Commit:12
+> 1 handling Ready
+  Ready MustSync=false:
+  HardState Term:2 Vote:3 Commit:12
+  CommittedEntries:
+  2/12 EntryNormal ""
+  Messages:
+  1->3 MsgAppResp Term:2 Log:0/12
+> 2 handling Ready
+  Ready MustSync=false:
+  HardState Term:2 Vote:3 Commit:12
+  CommittedEntries:
+  2/12 EntryNormal ""
+  Messages:
+  2->3 MsgAppResp Term:2 Log:0/12
+> 4 handling Ready
+  Ready MustSync=false:
+  HardState Term:2 Commit:12
+  CommittedEntries:
+  2/12 EntryNormal ""
+  Messages:
+  4->3 MsgAppResp Term:2 Log:0/12
+> 3 receiving messages
+  1->3 MsgAppResp Term:2 Log:0/12
+  2->3 MsgAppResp Term:2 Log:0/12
+  4->3 MsgAppResp Term:2 Log:0/12
+
+raft-state
+----
+1: StateFollower (Voter) Term:2 Lead:3
+2: StateFollower (Voter) Term:2 Lead:3
+3: StateLeader (Voter) Term:2 Lead:3
+4: StateFollower (Non-Voter) Term:2 Lead:3
+
+# ForgetLeader shouldn't affect the election timeout: if a follower
+# forgets the leader 1 tick before the election timeout fires, it
+# will still campaign on the next tick.
+set-randomized-election-timeout 2 timeout=3
+----
+ok
+
+tick-heartbeat 2
+----
+ok
+
+tick-heartbeat 2
+----
+ok
+
+forget-leader 2
+----
+INFO 2 forgetting leader 3 at term 2
+
+tick-heartbeat 2
+----
+INFO 2 is starting a new election at term 2
+INFO 2 became candidate at term 3
+INFO 2 [logterm: 2, index: 12] sent MsgVote request to 1 at term 3
+INFO 2 [logterm: 2, index: 12] sent MsgVote request to 3 at term 3
+
+stabilize
```

Comment:
ditto

Might be worth adding a `log-level=X` as a per-command option, i.e.

```
stabilize log-level=none
```

but no worries if you don't want to fry that fish now

Reply:
Yep, done.